variables.tf

```diff
@@ -93,6 +93,113 @@ variable "api_server_authorized_ip_ranges" {
+  type        = number
+  default     = 10
+}
+
+variable "auto_scaler_profile_expander" {
+  description = "The expander for the node group. Possible values are `random`, `most-pods`, `least-waste`, `price`, `priority`. Defaults to `random`."
+  type        = string
+  default     = "random"
+  validation {
+    condition     = contains(["least-waste", "most-pods", "priority", "random"], var.auto_scaler_profile_expander)
+    error_message = "Must be either \"least-waste\", \"most-pods\", \"priority\" or \"random\"."
+  }
+}
+
+variable "auto_scaler_profile_max_graceful_termination_sec" {
+  description = "The maximum time in seconds for which a node can be gracefully terminated. Defaults to 600."
+  type        = number
+  default     = 600
+}
+
+variable "auto_scaler_profile_max_node_provisioning_time" {
+  description = "The maximum time in seconds for which cluster autoscaler waits for node to be provisioned. Defaults to 15m."
+  type        = string
+  default     = "15m"
+}
+
+variable "auto_scaler_profile_max_unready_nodes" {
+  description = "The maximum number of unready nodes that CA tolerates. Defaults to 3."
+  type        = number
+  default     = 3
+}
+
+variable "auto_scaler_profile_max_unready_percentage" {
+  description = "The maximum percentage of unready nodes in a node group. Defaults to 45."
+  type        = number
+  default     = 45
+}
+
+variable "auto_scaler_profile_new_pod_scale_up_delay" {
+  description = "The delay after pod starts, before it's added to calculation of pending pods. It should be used together with skip_nodes_with_local_storage. Defaults to 10s."
+  type        = string
+  default     = "10s"
+}
+
+variable "auto_scaler_profile_scale_down_delay_after_add" {
+  description = "The delay after node addition, before another node can be removed. Defaults to 10m."
+  type        = string
+  default     = "10m"
+}
+
+variable "auto_scaler_profile_scale_down_delay_after_delete" {
+  description = "The delay after node deletion, before another node can be removed. Defaults to 10s."
+  type        = string
+  default     = "10s"
+}
+
+variable "auto_scaler_profile_scale_down_delay_after_failure" {
+  description = "The delay after node failure, before another node can be removed. Defaults to 3m."
+  type        = string
+  default     = "3m"
+}
+
+variable "auto_scaler_profile_scale_down_unneeded" {
+  description = "The time after which a node should be deleted after it has been unneeded for this long. Defaults to 10m."
+  type        = string
+  default     = "10m"
+}
+
+variable "auto_scaler_profile_scale_down_unready" {
+  description = "The time after which a node should be deleted after it has been unready for this long. Defaults to 20m."
+  type        = string
+  default     = "20m"
+
+}
+
+variable "auto_scaler_profile_scale_down_utilization_threshold" {
+  description = "The threshold in %!u(MISSING)nder which a node is considered for scale down. Defaults to 0.5."
+  type        = number
```

Comment:
According to the [document](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L201-L205) I would suggest that we set the type to `string` here along with the default value in `string`.